pkg/providers/amifamily/resolver.go

```diff
@@ -187,7 +183,7 @@ func (r Resolver) Resolve(ctx context.Context, nodeClass *v1beta1.EC2NodeClass,
 				DetailedMonitoring:  aws.BoolValue(nodeClass.Spec.DetailedMonitoring),
 				AMIID:               amiID,
 				InstanceTypes:       instanceTypes,
-				Licenses:            licenses,
+                Licenses:            nodeClass.Status.Licenses,
```

Comment:
@jonathan-innis What are your thoughts over this, should we use the resolved values from the nodeClass status? or load it again via the provider `r.licenseProvider.Get()`